.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: artifact-unit-tests
+on:
+  push:
+    paths:
+        - 'packages/artifact/**'
+  pull_request:
+    paths:
+        - 'packages/artifact/**'
+
+jobs:
+
+  build:
+    name: Build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest]
```

Comment:
Only testing on` ubuntu-latest` for now, I'm thinking one env should be sufficient

`macOS-latest` doesn't have `tsc` by default in the shell and with Windows some adjustments would have to be made to how env variables are set so.

Reply:
We should test on all environments, use `tsc` from your `node_modules` that you install and there are helpers in `core` and workflow commands to set environment variables.

Reply:
Took a bit to figure out why mac wasn't working. I had to add `typescript` to the dev dependencies for the artifact package because of the whole shell issue.